LandSandBoat/server
# Monster Follow Pathing

**_I affirm:_**

## What does this pull request do?

This is old code I wrote for Eden about 5 years ago. It adds a simple method for monsters to follow some target.

```lua
mob:follow(target, xi.follow.ROAM)
```

This is useful for many monsters such as:
 * Slave Globes for Mother Globe
 * Hpemde monster family
 * Lamia No.19 pets (almost any pet really)
 * Apkallu (they runaway from the player)

## Steps to test these changes

* Travel to Mother Globe with `!gotoid 17506396` and observe pathing
* You can kill the slaves out of order with `!hp 0` and they will properly adjust which mob they are following
* You can engage with Mother Globe and the slaves properly path the the character and after the character dies they resume their follow pathing

Additional steps taken from #3629 
1. Go to Al'taieu.
2. Kill a baby Ul'xzomit and respawn it. Observe it automatically re-attaches.
3. Kill a mother Ul'xzomit. Observe the baby Ul'xzomits start roaming by themselves.
4. Kill a baby Ul'xzomit and respawn it. Observe it starts roaming by itself.
5. Respawn the mother Ul'xzomit. Observe the babies start following it again.
6. Fight a mother Ul'xzomit. Kill a baby Ul'xzomit and respawn it. Observe the baby only links when it gets close.
7. Kill an Ul'xzomit and let it respawn normally. Observe its respawn time is now 5.5 minutes.
